plugins/builds/index.js

```diff
@@ -936,7 +985,13 @@ const buildsPlugin = {
                 if (+pid === current.pipeline.id) {
                     for (const nextJobName of Object.keys(pipelineJoinData[pid].jobs)) {
                         try {
+                            const resource = `pipeline:${current.pipeline.id}:event:${current.event.id}`;
```

Comment:
resource should be `nextPipelineId:nextJobName` if we do that, do we need any of the other changes ? since this will guarantee a sequential flow ?

Reply:
well point is that resources locked should be same for external event, for external event it's quite tricky to lock in format of nextPipelineId:nextJobName due to current implementation